src/tracks/SampleTrack.cpp

```diff
@@ -606,11 +942,158 @@ void SampleTCOView::paintEvent( QPaintEvent * pe )
 	p.end();
 
 	painter.drawPixmap( 0, 0, m_paintPixmap );
+	drawFades();
+}
+
+
+void SampleTCOView::enterEvent(QEvent *e)
+{
+	if (e != nullptr)
+	{
+		m_inside = true;
+	}
 }
 
 
+void SampleTCOView::leaveEvent(QEvent *e)
+{
+	if (e != nullptr)
+	{
+		m_inside = false;
+	}
+	TrackContentObjectView::leaveEvent(e);
+}
 
 
+void SampleTCOView::mouseMoveEvent(QMouseEvent* e)
+{
+	if ((e->buttons() & Qt::LeftButton) && (m_leftCorner || m_rightCorner))
+	{
+		float xpos = static_cast<float>(e->localPos().x());
+		float width = static_cast<float>(this->width());
+		float faderPos = qMin(qMax(0.0f, xpos / width), 1.0f);
+		if (m_leftCorner)
+		{
+			m_tco->m_sampleBuffer->setLeftFader(faderPos);
+		}
+		else if (m_rightCorner)
+		{
+			m_tco->m_sampleBuffer->setRightFader(faderPos);
+		}
+		updateSample();
+	}
+	else if ((e->buttons() & Qt::LeftButton) && (m_moveLeftP1 || m_moveLeftP2))
+	{
+		float xpos = e->localPos().x();
+		float ypos = qMin(qMax(e->localPos().y(),
+							   static_cast<qreal>(m_textLabelHeight)),
+						  static_cast<qreal>(this->height()));
```

Comment:
```suggestion
		float ypos = qMin(
			qMax(e->localPos().y(), static_cast<qreal>(m_textLabelHeight)),
			static_cast<qreal>(this->height())
		);
```
Also `qreal` by default is a `double` so this will lose precision on assigning to `float`.